apps/web/src/utils/importer.ts

```diff
@@ -17,100 +17,62 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { db } from "../common/db";
 import {
-  Note,
-  Notebook,
   ContentType,
-  LegacyNotebook
+  LegacyNotebook,
+  Note,
+  Notebook
 } from "@notesnook-importer/core/dist/src/models";
-import {
-  ATTACHMENTS_DIRECTORY_NAME,
-  NOTE_DATA_FILENAME
-} from "@notesnook-importer/core/dist/src/utils/note-stream";
-import { path } from "@notesnook-importer/core/dist/src/utils/path";
-import { type ZipEntry } from "./streams/unzip-stream";
-import { hashBuffer, writeEncryptedFile } from "../interfaces/fs";
 import { Notebook as NotebookType } from "@notesnook/core";
+import { db } from "../common/db";
+import { writeEncryptedFile } from "../interfaces/fs";
 
-export async function* importFiles(zipFiles: File[]) {
-  const { createUnzipIterator } = await import("./streams/unzip-stream");
-
-  for (const zip of zipFiles) {
-    let count = 0;
-    let filesRead = 0;
-
-    const attachments: Record<string, any> = {};
-
-    for await (const entry of createUnzipIterator(zip)) {
-      ++filesRead;
+export async function pushImportedNoteIntoDatabase(note: Note) {
+  const attachments: Record<string, EncryptedAttachmentFields> = {};
+  await processNoteAttachments(note.attachments, attachments);
+  await processNote(note, attachments);
+}
 
-      const isAttachment = entry.name.includes(
-        `/${ATTACHMENTS_DIRECTORY_NAME}/`
-      );
-      const isNote = !isAttachment && entry.name.endsWith(NOTE_DATA_FILENAME);
+type EncryptedAttachmentFields = {
+  chunkSize: number;
+  iv: string;
+  size: number;
+  salt: string;
+  alg: string;
+  key: {
+    key?: string;
+    salt?: string;
+  };
+};
```

Comment:
This should probably be a union of `Cipher` and `key`.

Reply:
I found a `Cipher` interface in the `crypto` package, but the fields didn't overlap fully. Currently I'm using this type definition, let me know if it is okay:

```ts
type EncryptedAttachmentFields = Awaited<
  ReturnType<typeof writeEncryptedFile>
> & {
  key: SerializedKey;
};
```

Reply:
Yeah that could work too, I guess.